tools/hermes-parser/js/flow-api-translator/src/flowToFlowDef.js

```diff
@@ -977,34 +978,116 @@ function convertClassDeclaration(
   ];
 }
 
-function convertSuperClass(
-  superClass: ?Expression,
-  superTypeParameters: ?TypeParameterInstantiation,
+function convertExpressionToIdentifier(
+  node: Expression,
   context: TranslationContext,
-): TranslatedResultOrNull<InterfaceExtends> {
-  if (superClass == null) {
-    return EMPTY_TRANSLATION_RESULT;
+): DetachedNode<Identifier> | DetachedNode<QualifiedTypeIdentifier> {
+  if (node.type === 'Identifier') {
+    return t.Identifier({name: node.name});
   }
 
-  if (superClass.type !== 'Identifier') {
-    throw translationError(
-      superClass,
-      `SuperClass: Non identifier super type of "${superClass.type}" not supported`,
-      context,
-    );
+  if (node.type === 'MemberExpression') {
+    const {property, object} = node;
+    if (property.type === 'Identifier' && object.type !== 'Super') {
+      return t.QualifiedTypeIdentifier({
+        qualification: convertExpressionToIdentifier(object, context),
+        id: t.Identifier({name: property.name}),
+      });
+    }
   }
+
+  throw translationError(
+    node,
+    `Expected ${node.type} to be an Identifier or Member with Identifier property, non-Super object.`,
+    context,
+  );
+}
+
+function convertSuperClassHelper(
+  detachedId: DetachedNode<Identifier | QualifiedTypeIdentifier>,
+  nodeForDependencies: ESNode,
+  superTypeParameters: ?TypeParameterInstantiation,
+  context: TranslationContext,
+): TranslatedResultOrNull<InterfaceExtends> {
   const [resultTypeParams, typeParamsDeps] =
     convertTypeParameterInstantiationOrNull(superTypeParameters, context);
-  const superDeps = analyzeTypeDependencies(superClass, context);
+  const superDeps = analyzeTypeDependencies(nodeForDependencies, context);
   return [
     t.InterfaceExtends({
-      id: asDetachedNode(superClass),
+      id: detachedId,
       typeParameters: resultTypeParams,
     }),
     [...typeParamsDeps, ...superDeps],
   ];
 }
 
+function convertSuperClass(
+  superClass: ?Expression,
+  superTypeParameters: ?TypeParameterInstantiation,
+  context: TranslationContext,
+): TranslatedResultOrNull<InterfaceExtends> {
+  if (superClass == null) {
+    return EMPTY_TRANSLATION_RESULT;
+  }
+
+  switch (superClass.type) {
+    case 'Identifier': {
+      return convertSuperClassHelper(
+        asDetachedNode(superClass),
+        superClass,
+        superTypeParameters,
+        context,
+      );
+    }
+    case 'MemberExpression': {
+      return convertSuperClassHelper(
+        convertExpressionToIdentifier(superClass, context),
+        superClass,
+        superTypeParameters,
+        context,
+      );
+    }
+    case 'TypeCastExpression': {
```

Comment:
adding `AsExpression` case here causes a bunch of flow errors -- didn't look into it further